### ISSUE TITLE
utils.rb:435:in `finalize_source': warning: URI.unescape is obsolete

### ISSUE THREAD

**Issue by vestedpr-dev** [cloudinary/cloudinary_gem#315]
Thank you for developing Cloudinary.  It's very useful for my team and I.

Using Ruby 2.4.4, when running my tests, I see this warning:

```
/.rbenv/versions/2.4.4/lib/ruby/gems/2.4.0/gems/cloudinary-1.9.1/lib/cloudinary/utils.rb:435:in `finalize_source': warning: URI.unescape is obsolete
```

In my application, I have been able to replace URI.escape with ::Addressable::URI.encode, although, I doubt you would want to introduce a dependency for this.

**Comment by marissa-masangcay** [comment]
Hi @vestedpr-dev, thank you for sharing this information. We're so glad to hear you're enjoying our product! I'll be sure to pass this info on to our dev team for further evaluation.

Cheers :)

**Comment by rubydev** [comment]
Hello, any new info about this warning?

**Comment by aleksandar-cloudinary** [comment]
Hi @rubydev - We're in the process of rewriting all of our SDK code into a second iteration with shared code standards, consolidation of dependencies, with focus on easier extensibility and this warning would be fixed automatically as part of that project. For example, one of the first SDKs under this new version is our PHP SDK which you can find here - https://github.com/cloudinary/cloudinary_php_v2

**Comment by aleksandar-cloudinary** [comment]
Hi @rubydev - We're in the process of rewriting all of our SDK code into a second iteration with shared code standards, consolidation of dependencies, with focus on easier extensibility through object orientation and more. This warning would be fixed automatically as part of that project. For example, one of the first SDKs under this new SDK 2.0 project is our the PHP one which you can find here - https://github.com/cloudinary/cloudinary_php_v2

**Comment by tbsvttr** [comment]
I get a similar warning with Ruby 2.7.1, Rails 6.0.2.2 and Cloudinary 1.13.1.

>   /app/vendor/bundle/ruby/2.7.0/gems/cloudinary-1.13.1/lib/cloudinary/utils.rb:566: warning: URI.unescape is obsolete

I get this logged to my log so often that the log becomes useless and exceeds quotas. Is there a fix for this?

**Comment by ThomasConnolly** [comment]
Same here. Ruby 2.7.1 Rail 6.0.2.2.
Cloudinary used to be so good with Rails. The Attachinary gem was fantastic but nobody has touched it in years.

**Comment by idobarnoam** [comment]
Hey, @tbsvttr @ThomasConnolly Like mentioned this is planned to be handled in our Ruby 2.0 SDK that is currently in the works. With that said, we are looking into this specifically to see how this could be handled in the current SDK.

We'll keep this thread updated with any new insights about this.

**Comment by TImPDaley** [comment]
Hi Put me on the list too... Ruby 2.7.0 Rails 6.0.2.2 Cloudinary please.

**Comment by urchymanny** [comment]
two years and this is still open?

**Comment by idobarnoam** [comment]
Hi @urchymanny 
This is actually in progress from our side.
When this is done we would make sure to update the thread

I hope that helps.

**Comment by momoip** [comment]
Hi,
This warning is fixed in the Ruby SDK 2.0.

Cheers.
